src/arkode/arkode_lsrkstep_io.c

```diff
@@ -0,0 +1,656 @@
+int LSRKStepSetMaxNumStages(void* arkode_mem, int stagemaxlimit)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  if (stagemaxlimit < 2)
+  {
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "stagemaxlimit must be greater than or equal to 2");
+    return (ARK_ILL_INPUT);
+  }
+
+  step_mem->stagemaxlimit = stagemaxlimit;
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  LSRKStepSetDomEigSafetyFactor sets the safety factor for the DomEigs.
+  ---------------------------------------------------------------*/
+int LSRKStepSetDomEigSafetyFactor(void* arkode_mem, sunrealtype domeigsfty)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  if (domeigsfty < SUN_RCONST(1.0))
+  {
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "domeigsfty must be greater than or equal to 1");
+    return (ARK_ILL_INPUT);
+  }
+
+  step_mem->domeigsfty = domeigsfty;
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  LSRKStepSetSSPStageNum sets the number of stages in the following
+  SSP methods:
+
+      ARKODE_LSRK_SSPs_2  -- numofstages must be greater than or equal to 2
+      ARKODE_LSRK_SSPs_3  -- numofstages must be a full-square greater than or equal to 9
+      ARKODE_LSRK_SSP10_4 -- numofstages must be equal to 10 - no need to call!
+
+  This set routine must be called after calling LSRKStepSetMethod with an SSP method
+  ---------------------------------------------------------------*/
+int LSRKStepSetSSPStageNum(void* arkode_mem, int numofstages)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  switch (step_mem->LSRKmethod)
+  {
+  case ARKODE_LSRK_SSPs_2:
+    if (numofstages < 2)
+    {
+      arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                      "numofstages must be greater than or equal to 2");
+      return (ARK_ILL_INPUT);
+    }
+    step_mem->reqstages = numofstages;
+    break;
+
+  case ARKODE_LSRK_SSPs_3:
+    if (numofstages < 9 ||
+        (ceil(SUNRsqrt(numofstages)) != floor(SUNRsqrt(numofstages))))
```

Comment:
That cost me lots of commits and still returned test failures. It was also hard to debug since everything seemed fined in my local machine. I just had to revert it to the original form to avoid test failuires. If you want to look at what was wrong with my code that would be awesome.